WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,106 @@
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    private void tapLikeButton() {
+        mSwipeForMore.waitUntilGone(DEFAULT_TIMEOUT);
+        // Even though it was working locally in simulator, tapping the footer buttons,
+        // like 'mLikeButton.click()', was not working in CI.
+        // The current workaround is to use arrows navigation.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_DOWN);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_CENTER);
+        resetFocus();
+    }
+
+    private void resetFocus() {
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+    }
+
+    public ReaderPage goBackToReader() {
+        mDevice.pressBack();
+
+        return new ReaderPage();
+    }
+
+    public ReaderViewPage slideToPreviousPost() {
+        swipeToRight();
+
+        return this;
+    }
+
+    public ReaderViewPage slideToNextPost() {
+        swipeToLeft();
+
+        return this;
+    }
+
+    public ReaderViewPage verifyPostDisplayed(String title) {
+        assertTrue("Post title was not displayed. Target post: " + title, isTextDisplayed(title));
+
+        return this;
+    }
+
+    public ReaderViewPage verifyPostLiked() {
+        boolean isLiked = mDevice
+                .findObject(new UiSelector().textContains("You like this.")).waitForExists(DEFAULT_TIMEOUT);
+
+        assertTrue("Liker was not displayed.", isLiked);
+
+        return this;
+    }
+
+    public ReaderViewPage verifyPostNotLiked() {
+        boolean likerDisplayed = mLikerContainer.exists();
+
+        assertFalse("Liker faces container was displayed.", likerDisplayed);
```

Comment:
Think you mentioned this previously as well and it's not the first time I'm seeing this but got confused for a second. It's still a bit weird to read the error first 😅